system/lsan/lsan_main.c

```diff
@@ -0,0 +1,38 @@
+/****************************************************************************
+ * apps/system/lsan/lsan_main.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <stdio.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+extern void __lsan_do_recoverable_leak_check(void);
```

Comment:
where does this come from? more commits incoming?

Reply:
see the Summary

Reply:
thanks for the update. so this is an extenstion to GCC? what about other compilers? if only available to GCC this should be noted both in Kconfig comments and Documentation.

Reply:
this depends SIM_ASAN, if this supports, then this tool working

The Kconfig file has well depends

Reply:
clang also support.